endpoints.go

```diff
@@ -1,22 +1,39 @@
-package dothill
+package exosx
 
 import (
 	"crypto/md5"
+	"crypto/sha256"
 	"fmt"
 	"strings"
 )
 
 // Login : Called automatically, may be called manually if credentials changed
-func (client *Client) Login() error {
+func (client *Client) Login(hashf ...string) error {
 	userpass := fmt.Sprintf("%s_%s", client.Username, client.Password)
-	hash := md5.Sum([]byte(userpass))
-	res, _, err := client.FormattedRequest("/login/%x", hash)
+
+	var hashmethod string = "md5"
+	if len(hashf) > 0 {
+		hashmethod = hashf[0]
+	}
+
+	var hash string
+	if hashmethod == "md5" {
+		hash = fmt.Sprintf("%x", md5.Sum([]byte(userpass)))
+	} else if hashmethod == "sha256" {
```

Comment:
I think you're already aware of this, but there's no need to support SHA256; all generations of the array support either MD5 (in the URL) or Basic Auth  (in the HTTP headers) and you can simply send both at the same time to cover all bases.  At some point we'll have to deprecate SHA256 because it's equivalent to putting the username in password in the URL, but MD5 will still be OK because it's ignored by the newer firmware that supports Basic Auth.

Reply:
I updated the code to use this method that you suggested. The code uses both SetBasicAuth() and md5 and removed older changes. I also tested successfully against older and new API versions.

Reply:
I updated the code to use SetBasicAuth() and removed the use of SHA256. Tests passed with older and newer API versions.